zcash_primitives/src/consensus.rs

```diff
@@ -5,22 +5,28 @@ use std::fmt;
 
 /// Zcash consensus parameters.
 pub trait Parameters {
-    fn activation_height(nu: NetworkUpgrade) -> Option<u32>;
+    fn activation_height(&self, nu: NetworkUpgrade) -> Option<u32>;
 
-    fn is_nu_active(nu: NetworkUpgrade, height: u32) -> bool {
-        match Self::activation_height(nu) {
+    fn is_nu_active(&self, nu: NetworkUpgrade, height: u32) -> bool {
+        match self.activation_height(nu) {
```

Comment:
AFAICT `MainNetwork` and `TestNetwork` are still marker structs that don't need to be instantiated, so there is nothing to access from `&self`. Are there other changes in this PR require adding `&self` to the `Parameters` methods? If not, revert these.